### PR TITLE
React-docgen version upgraded to v3 to support Flow annotations

### PR DESCRIPTION
**Pull request by danielfx90**
# Problem
`<*>` tokens throw syntax error. For example when doing:
```
import type { Children } from "react";
...
type Props = {
   children?: Children<*>,
...
```

# Solution
`react-docgen` package version upgraded from v2 to v3: https://github.com/reactjs/react-docgen/issues/181

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "ast-types": "^0.9.11",
+    "babylon": "^7.0.0-beta.8",
```
Comment (sapegin) [inline package.json]: Babylon is a dependency of react-docgen, why is it here?

File: package.json [hunk 2]
```diff
@@ -42,6 +43,7 @@
     "es6-object-assign": "~1.1.0",
     "es6-promise": "^4.1.0",
     "escodegen": "^1.8.1",
+    "eslint": "^3.19.0",
```
Comment (sapegin) [inline package.json]: Should stay in devDependencies.

File: package.json (hunk 1 above)
Comment (danielfx90) [inline package.json]: Sorry, I was trying something and I forgot to remove it. Thanks!

File: package.json (hunk 2 above)
Comment (danielfx90) [inline package.json]: True that!

**Comment by danielfx90** [comment]
@sapegin all done!

**Comment by danielfx90** [comment]
@sapegin Sorry, still getting the hang of it. Now all three checks have passed.

**Comment by sapegin** [comment]
Merged, thanks!

**Comment by sapegin** [comment]
Small issue with this change: https://github.com/nerdlabs/react-docgen-displayname-handler/blob/master/package.json#L43

```
npm ERR! peer dep missing: react-docgen@2.x, required by react-docgen-displayname-handler@1.0.0
```

Not sure if it breaks anything, if not probably worth sending a PR to react-docgen-displayname-handler to relax required react-docgen version.

**Comment by danielfx90** [comment]
I'll check it out!

**Pull request merged**